src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -127,19 +148,49 @@ public OutputStream getOutputStream() {
         }
     }
 
+    @RequestMapping(value = "/subscriptions/{subscription_id}/events", method = RequestMethod.POST)
+    public StreamingResponseBody streamEvents(
+            @PathVariable("subscription_id") final String subscriptionId,
+            @Valid @RequestBody final UserStreamParameters userParameters,
+            final HttpServletRequest request,
+            final HttpServletResponse response,
+            final Client client)
+            throws UnprocessableEntityException {
+
+        final StreamParameters streamParameters = StreamParameters.of(userParameters,
+                nakadiSettings.getDefaultCommitTimeoutSeconds(), client);
+
+        return stream(subscriptionId, request, response, client, streamParameters);
+    }
+
     @RequestMapping(value = "/subscriptions/{subscription_id}/events", method = RequestMethod.GET)
     public StreamingResponseBody streamEvents(
             @PathVariable("subscription_id") final String subscriptionId,
-            @RequestParam(value = "max_uncommitted_events", required = false, defaultValue = "10")
-            final int maxUncommittedSize,
-            @RequestParam(value = "batch_limit", required = false, defaultValue = "1") final int batchLimit,
+            @Nullable @RequestParam(value = "max_uncommitted_events", required = false) final Integer
+                    maxUncommittedEvents,
+            @Nullable @RequestParam(value = "batch_limit", required = false) final Integer batchLimit,
             @Nullable @RequestParam(value = "stream_limit", required = false) final Long streamLimit,
-            @RequestParam(value = "batch_flush_timeout", required = false, defaultValue = "30") final int batchTimeout,
+            @Nullable @RequestParam(value = "batch_flush_timeout", required = false) final Integer batchTimeout,
             @Nullable @RequestParam(value = "stream_timeout", required = false) final Long streamTimeout,
-            @Nullable
-            @RequestParam(value = "stream_keep_alive_limit", required = false) final Integer streamKeepAliveLimit,
+            @Nullable @RequestParam(value = "stream_keep_alive_limit", required = false) final Integer
+                    streamKeepAliveLimit,
             final HttpServletRequest request, final HttpServletResponse response, final Client client)
-            throws IOException {
+            throws IOException, UnprocessableEntityException {
```

Comment:
I'm not sure where IOException and UnprocessableEntityException are thrown, and what is the difference with the previous method, which only throws UnprocessableEntityException?

Reply:
Yes, these exceptions are not thrown actually, it remained from some old times. I removed that now https://github.com/zalando/nakadi/pull/848/commits/220a7106992d5125c9f027b29001b02ea750f7d4